packages/catalog-process/src/services/catalogService.ts

```diff
@@ -129,11 +129,13 @@ const retrieveDocument = (
   descriptor: Descriptor,
   documentId: EServiceDocumentId
 ): Document => {
-  const doc = descriptor.docs.find((d) => d.id === documentId);
-  if (doc === undefined) {
+  const document = [...descriptor.docs, descriptor.interface].find(
+    (doc) => doc != null && doc.id === documentId
```

Comment:
The `doc != null` could be removed, there should be no need that check
```suggestion
    (doc) => doc.id === documentId
```

Reply:
The `interface` might be undefined, so in that case `id` can't be read.

Reply:
Ops, you're right! I thought that you were spreading `descriptor.interface` into the array for some reasons...